include/mbedtls/ssl_internal.h

```diff
@@ -310,7 +310,7 @@ struct mbedtls_ssl_handshake_params
 #endif
 
 #if defined(MBEDTLS_SSL_ASYNC_PRIVATE)
-    int async_in_progress : 1;          /*!< an asynchronous operation is in progress */
+    unsigned int async_in_progress : 1; /*!< an asynchronous operation is in progress */
```

Comment:
Haha, I almost commented about that, because I remember when I checked the pitfalls of bitfields I came to the conclusion that only `unsigned` was safe for what I wanted to do, but I didn't comment because I didn't remember the reason (and assumed you knew that kind of thing better than me, which in retrospect is an obviously counter-productive assumption to make while reviewing). Good that gcc came to the rescue, and next time I'm in doubt about a C feature I'll ask if you're sure.